cosmos/lib/evm.go

```diff
@@ -42,6 +48,10 @@ func DeployOnEVMFromPrecompile(
 	contractCode string, // hex-encoded string
 	constructorArgs ...any,
 ) (common.Address, []byte, error) {
+	if utils.MustGetAs[precompile.MultiStore](ctx.MultiStore()).IsReadOnly() {
+		return common.Address{}, nil, vm.ErrWriteProtection
```

Comment:
realized bc we return an error here, this will actually get wrapped with `ErrExecutionReverted` which could be wrong

Reply:
ez 
![image](https://github.com/berachain/polaris/assets/113940816/0e68d7f2-8657-4a56-b378-96e77b196ed4)

Reply:
fixed?

Reply:
yes